src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProviderResponseValidation.java

```diff
@@ -31,6 +31,11 @@ public class EmbeddingProviderResponseValidation implements ClientResponseFilter
   @Override
   public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext)
       throws JsonApiException {
+    // Check if the header is null
```

Comment:
When timed out, is it not throwing any http error code? Should we run this filter only when the response http status code is in 2xx?

Reply:
I would think true timeout would throw an exception and not server `null` headers... so it almost seems like an odd error condition by rest client we use.